power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -147,22 +147,43 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
 
     struct UpdateCompProperties {
         std::string name{};
-        bool has_id{false};           // if the component has id
-        bool ids_all_na{false};       // if all ids are all NA
-        bool ids_part_na{false};      // if some ids are NA but some are not
-        bool uniform{false};          // if the component is uniform
-        bool is_columnar{false};      // if the component is columnar
-        bool ids_match{false};        // if the ids match
-        Idx elements_ps_in_update{0}; // count of elements for this component per scenario in update
-        Idx elements_in_base{0};      // count of elements for this component per scenario in input
-        constexpr bool no_id_col() const { return is_columnar && (!has_id || ids_all_na); }
-        constexpr bool no_id_row() const { return !is_columnar && (!has_id && ids_all_na); }
-        constexpr bool qualify_for_optional_id() const { return ids_match && ids_all_na && !ids_part_na; }
-        constexpr bool provided_ids_valid() const { return has_id && ids_match && !ids_all_na && !ids_part_na; }
-        constexpr bool is_empty_component() const { return !has_id && ids_all_na; }
+        bool has_any_elements{false}; // whether the component has id
```

Comment:
Please revert the renaming; this is clearly a wrong choice

Reply:
comment was deprecated